src/TaskLineRenderer.ts

```diff
@@ -31,6 +31,53 @@ export const LayoutClasses: { [c in TaskLayoutComponent]: string } = {
     blockLink: 'task-block-link',
 };
 
+type AttributeValueCalculator = (component: TaskLayoutComponent, task: Task) => string | null;
+
+export class FieldLayoutDetail {
+    className: string | null;
+    attributeName: string;
+    attributeValueCalculator: AttributeValueCalculator;
+
+    constructor(className: string | null, attributeName: string, attributeValueCalculator: AttributeValueCalculator) {
```

Comment:
From the perspective of someone maintaining this, it's a bit confusing that `className` can be `string | null` but:

- there are no uses of `null`
- there is no JSDocs to explain why one might to pass `null` in.

Reply:
The empty string seems to be a special value of `attributeName`. Please add some JSDocs to explain why.

Reply:
Also, IIRC there was some kind of correlation between `attributeName` and `attributeValueCalculator`.

If we don't get this kind of knowledge written down when the code is added, time has to get spent re-learning this stuff later on in maintenance, so it's much better to explain it in comments at the point when it's written...

Reply:
> From the perspective of someone maintaining this, it's a bit confusing that `className` can be `string | null`...:

Actually, now that `blockId` has a class name, I just removed the possibility for class name to be null. This will simplify other aspects of the code.

Same with the return of `AttributeValueCalculator`.

> The empty string seems to be a special value of `attributeName`. Please add some JSDocs to explain why.

I also added a throw on that since in this case to have a runtime error with more information.

Reply:
Thanks for documenting the constructor of `FieldLayoutDetail`. Resolving.